### ISSUE TITLE
NGINX SSL directive should be aligned to new format

### ISSUE THREAD

**Issue by Hooverdan96** [rockstor/rockstor-core#2275]
In some cases there seems to be an issue with the deprecated SSL directive in nginx.conf being commented out and the niginx service subsequently not enabling SSL for the WebUI. See this forum entry for the symptoms and interim resolution on a CentOs-based installation (tbd whether this can be a problem on the rebased OpenSUSE.

https://forum.rockstor.com/t/solved-power-loss-reboot-successful-but-no-webui-access/7608/14

This should be investigated further and possibly changed so the file is consistent with the new format during initial installation of Rockstor.

**Comment by phillxnet** [comment]
> This should be investigated further and possibly changed so the file is consistent with the new format during initial installation of Rockstor.

@Hooverdan96 Agreed. And I think we should now pick this one up.
Current testing, when build on and for openSUSE Leap 15.3 has:
```
2022/06/15 17:10:51 [warn] 25141#25141: the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /opt/rockstor/etc/nginx/nginx.conf:47
2022/06/17 15:49:17 [warn] 13123#13123: the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /opt/rockstor/etc/nginx/nginx.conf:47
2022/06/17 17:49:37 [warn] 7179#7179: the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /opt/rockstor/etc/nginx/nginx.conf:47
2022/06/17 17:52:23 [warn] 10347#10347: the "ssl" directive is deprecated, use the "listen ... ssl" directive instead in /opt/rockstor-dev/etc/nginx/nginx.conf:47
```

Pretty sure all we need here is an update of our template file here:
https://github.com/rockstor/rockstor-core/blob/master/conf/nginx.conf.in

Although we may end up programmatically editing that in some places that for example change the Web-UI port. And initrock should be checked just in case as we invoke a service reconfig via:
https://github.com/rockstor/rockstor-core/blob/81a0d5f5e0a6dfe5a922199828f66eeab0253e65/src/rockstor/scripts/initrock.py#L125-L130

**Comment by Hooverdan96** [comment]
https://github.com/rockstor/rockstor-core/blob/81a0d5f5e0a6dfe5a922199828f66eeab0253e65/conf/nginx.conf.in#L44-L50

looks like Line 45 has to only be changed to: 
<code> listen 443 ssl default_server;</code> and line 47 can be removed (or commented out).

**Comment by phillxnet** [comment]
Noting that the deprecation messages in question, re the ssl directive, can be viewed by:
System -> Log manager -> upper "Logs Reader" section -> select "Nginx stderr (WebUI)" -- click Read log

**Comment by phillxnet** [comment]
Re:

> Pretty sure all we need here is an update of our template file here: https://github.com/rockstor/rockstor-core/blob/master/conf/nginx.conf.in

After recent clean-ups in our testing (read development) branch, this nginx.conf template file is now located here:
https://github.com/rockstor/rockstor-core/blob/testing/etc/nginx/nginx.conf

and since we have now dropped the pre-processing associated with our now removed buildout (replace by Poetry) this file is used, as-is via an nginx systemd override file here:

https://github.com/rockstor/rockstor-core/blob/testing/conf/30-rockstor-nginx-override.conf

which enables (via override) our config test and enablement via:

https://github.com/rockstor/rockstor-core/blob/28dc8bee74fb0d5276bd9d119b238c463e906adc/conf/30-rockstor-nginx-override.conf#L7-L11

which is, in turn, enacted within initrock.

**Comment by FroggyFlox** [comment]
Closing as fixed by #2465
